### DISCUSSION TITLE
Can not create filter to filter by office

### DISCUSSION THREAD

**Discussion by meloforroty** [microsoft-search/pnp-modern-search#3904] [Q&A]
I am trying to create a filter for my company directory page. I have already set up the PnP search bar, search results, and filter webparts. Moreover, I have set up the filter webpart to be able to filter by department. I made sure to go through each user in Entra ID and added the department they belong to. I am trying to create a second filter so that you can filter the company directory by company however even though I have gone through Entra ID and added the office for each user, when I go to add filter and put "BaseOfficeLocation" for the filter field, when I hit save and apply, the filter does not show up in my intranet page. I have checked through the SharePoint admin center that this property is refinable, queryable, and retrieveable but still it does not work when I try to set up the filter in my SharePoint intranet. Please help me with this.

**Comment by kasperbolarsen** [comment]
In order for the refiner to show up there are some requirements:
1) some users must have a value in that manage property. Adding the value in Entra isn't enough, it must be synced to the SharePoint User Profile Application. You can check if any of the users have a value is that field using the SP editor or SharePoint Search query tool
2) Each and every Managed Property used for filtering must be added to the "Selected Property" list in the Results web part 

PS. you are on the right track :-)

**Comment by meloforroty** [comment]
Thank you for your comment. I went to manage user profiles in the SharePoint admin center and the "Office" field is populated correctly for every user. I can not find "BaseOfficeLocation" or "Office" to check off in the Selected Property list in the Results web part. I currently have checked off: 

"AuthorOWSUSER", "contentclass", "contentTypeId", "Created", "CreatedBy", "DefaultEncodingURL", "FileType", "HitHighlightedProperties", "HitHighlightedSummary","HtmlFileType", "JobTitle", "NormListID", "NormSiteID", "NormUniqueID", "NormWebID", "owstaxidmetadataalltagsinfo, "Path", "PictureThumbnailURL", "SiteLogo", "SiteTitle", "SPSiteURL", "SPWebUrl", "Title", "UniqueID", "UserName", "WorkPhone"

Please let me know if I am missing something. Thanks for your help!

**Comment by kasperbolarsen** [comment]
The list in Selected Properties is not complete,  and in this case you will have to type BaseOfficeLocation in yourself

**Comment by meloforroty** [comment]
I typed in BaseOfficeLocation and when I go to customize filters in the Search filter webpart, it still does not work.

**Comment by meloforroty** [comment]
Here is how I have set up the filters in the filter webpart (see link below). The Department one works but the Office one does not work.

Link: https://imgur.com/8VsLmun

**Comment by kasperbolarsen** [comment]
have you checked if any of the users have a BaseOfficeLocation value assigned? I recommend using either the debug layout or even better, SP Editor / SharePoint Search query tool

**Comment by meloforroty** [comment]
Yes I checked and all users have a BaseOfficeLocation assigned.

**Comment by kasperbolarsen** [comment]
Using which of the tools ? Checking the UPA isn't sufficient

**Comment by meloforroty** [comment]
I checked it using the SharePoint Search Query Tool. I also confirmed this by going into the SharePoint Admin center and checking the properties of each profile and it shows that the Office field is filled with the correct office values which I assigned. When I go to the PnP webparts and go to the query template and type "{searchTerms} Miami Developments:" it shows all and only the people who are part of this company. The same happens when I go and click "{searchTerms} S.E.M. Construction:'", How do I add this as a filter in the same manner which the department filter is set up (see attachment)

Link: https://imgur.com/a/APnmVFk

**Comment by kasperbolarsen** [comment]
Can you verify that the query {searchTerms} BaseOfficeLocation:"Miami Developments" yields a result? It should, and that is the precondition for making it work as a Refiner. Once that have been confirmed, you should ensure that BaseOfficeLocation is selected in "Selected Properties" , and add BaseOfficeLocation in the Refiner web part ( I assume that you have connected the Refiner web part to the Results web part, AND the Results web part to the Refiner web part )
Have fun with search :-)

**Comment by meloforroty** [comment]
Somehow when I search up {searchTerms} BaseOfficeLocation: "Miami Developments" and {searchTerms} BaseOfficeLocation: "S.E.M. Construction" it returns 0 results. However, when I search up {searchTerms} S.E.M. Construction: and {searchTerms} Miami Developments: it yields the proper results. Please note that BaseOfficeLocation was selected for all of the above queries. Perhaps this means that there are no values for BaseOfficeLocation and I was wrong. Is there something else I am missing. Thank you for your continued guidance.
